app/controllers/approvals_controller.rb

```diff
@@ -6,7 +6,9 @@ def create
 
     current_student.add(@approvable)
 
-    @graduated = current_student.total_credits >= 450 && previous_total_credits < 450
+    group_credits_met = SubjectGroup.all.all? { |group| current_student.group_credits_met?(group) }
+
+    @graduated = current_student.total_credits >= 450 && previous_total_credits < 450 && group_credits_met
```

Comment:
thoughts about creating a method in `BaseStudent` for this, so we can test it?

also wonder if we should also take into account the `group_credits_met ` before adding the approvable, so we can correctly determine the user was not graduated before and they are now

Reply:
I moved that logic to the model in 20a336b43732fe7a167100e67a117380e1d116bf